### PR TITLE
Switch to cgr.dev/chainguard/static image + remove symbol + DWARF table

### PR DESCRIPTION
**Pull request by marcofranssen**
Switch to chainguard static image + remove symbol and DWARF table

### CODE REVIEW

**Comment by azdagron** [comment]
The name of this PR is misleading. With GCO_ENABLED=0, the binary should already have no library dependencies. The `-s -x` ldflags simply reduce the size of the binary by omitting extra symbol and DWARF metadata.

**Comment by marcofranssen** [comment]
> The name of this PR is misleading. With GCO_ENABLED=0, the binary should already have no library dependencies. The `-s -x` ldflags simply reduce the size of the binary by omitting extra symbol and DWARF metadata.

Renamed

File: Dockerfile [hunk 1]
```diff
@@ -28,12 +28,11 @@ COPY --link --from=xx / /
...
-    go build -o bin/spire-controller-manager main.go
+    go build -ldflags '-s -w' -o bin/spire-controller-manager main.go
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-#FROM gcr.io/distroless/static:nonroot
-FROM gcr.io/distroless/base AS spire-controller-manager
+# See https://github.com/chainguard-images/images/tree/main/images/static
+# Used to run static binaries, includes tzdata and ca-certificates
+FROM cgr.dev/chainguard/static:latest AS spire-controller-manager
```
Comment (azdagron) [inline Dockerfile:35]: What benefits do we bring in by switching to this image over the base distroless image?

Comment (marcofranssen) [reply Dockerfile:35]: Static image has ca-certificates and tzdata patched on a daily basis.

Furthermore it is build on Wolfi-OS (Linux Undistro) which aims to be memory safe, having SBOMs by default. E.g. we could install ca-certificates and tzdata in a builder image and copy it here, using this image we don't need to as Chainguard is having daily builds for these images including security checks against sboms and CVEs.

**Comment by azdagron** [comment]
This is fine, except that spire-controller-manager needs to run as root (for now, so it can connect to the UDS created by spire-server as root) and the chainguard image runs as a non-root user (this is why the CI/CD tests are failing).

**Comment by marcofranssen** [comment]
Will check to make some PRs soon so al spire images can be ran as non root. We are also looking into this on the helm charts.

**Comment by azdagron** [comment]
Closing for now. Once we have a good story for running non-root, maybe we can revisit this.

**Pull request closed without merging**